fml/memory/weak_ptr.h

```diff
@@ -43,7 +48,7 @@ class WeakPtr {
   WeakPtr() : ptr_(nullptr) {}
 
   // Copy constructor.
-  WeakPtr(const WeakPtr<T>& r) = default;
+  explicit WeakPtr(const WeakPtr<T>& r) = default;
```

Comment:
@iskakaushik Changed after your last approval:
I found implicit casting could be problematic as someone could want a TaskRunnerAffineWeakPtr but end up getting a WeakPtr.

Reply:
LGTM